File: backend/src/models/terrain.py


from typing import Optional
from pydantic import BaseModel


class Terrain(BaseModel):
    id: Optional[str] = None
    created_at: Optional[str] = None
    name: str
    game_id: str
    texture: str
    z_index: int = 0
    solid: bool = False
    opacity: float = 1.0
    rotation: int = 0

File: frontend/src/models/terrain.ts
import { Realm } from "../game/world"

export interface Terrain {
    id: string
    created_at: string
    name: string
    game_id: string
    texture: string
    x: number
    y: number
    z: number
    realm: Realm
    solid: boolean
    opacity: number
    rotation: number
    ext?: string
    variant: string
    variants: string[]
}
